src/controllers/ModelsController.ts

```diff
@@ -90,8 +90,18 @@ const makeModelsController = (
         _id: attrs.signingKeyId,
         radiksType: 'SigningKey',
       });
-      const message = `${attrs._id}-${attrs.updatedAt}`;
-      if (verifyECDSA(message, publicKey, req.query.signature)) {
+      const {
+        signature,
+        updatedAt,
+      }: { signature: string; updatedAt: string } = req.query;
+      if (!updatedAt || parseInt(updatedAt, 10) <= attrs.updatedAt) {
```

Comment:
This may be an ignorant question --
I can't tell if the signature validation encompasses this `updatedAt` value, is the value from the client already validated before this function? 
As in, is there a server-side component preventing a malicious client from just setting this value to `Number.MAX_SAFE_INTEGER` or something like that?

Reply:
There is no such check for `updatedAt`. We should add a validator check that the value is near the server's time.